src/convertToObject.js

```diff
@@ -6,7 +6,22 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const array = sourceString.split(';');
+
+  const object = array.reduce((accumulator, current) => {
+    const element = current.split(':');
+
+    if (element.length === 2) {
+      const property = element[0].trim();
+      const value = element[1].trim();
+
+      accumulator[property] = value;
+    }
+
+    return accumulator;
+  }, {});
```

Comment:
[CODE KNOWLEDGE] - There is no need to make a copy before each operation. Moreover, methods like `reduce` don't mutate the original array. So you can directly use `sourceString.split(';').reduce(...)`.